rfcs/0001-masonry-backend.md

```diff
@@ -0,0 +1,211 @@
+
+### `xilem`
+
+The element type in `xilem` is bound to the `masonry::Widget` trait. Different views defined in Xilem have matching Widget types defined in Masonry, though the correspondence isn't always one-to-one.
+
+`Widget` in Masonry is a trait that represent units of graphical composition, things like text and buttons and boxes and containers. To implement Widget, a type must:
+
+- Be `'static`.
+- Provide info about its children.
+- Handle user events.
+- Compute its layout.
+- Draw itself in a paint target.
+
+See the [Widget trait definition](https://github.com/PoignardAzur/masonry-rs/blob/3cfc2cddec05d48a31a9812bdda27792d336f03f/src/widget/widget.rs#L65-L199) for more details.
+
+
+### Directory structure
+
+The `https://github.com/linebender/xilem` repository has the following crates in its `crates/` directory:
+
+- `xilem_core/`
+- `xilem_web/`
+- `masonry/`
+
+The main `xilem/` crate is defined at the project root.
+
+## Implementation strategy
+
+The implementation will require several steps:
+
+- Refactor Masonry to use the same dependencies Xilem currently uses, especially Glazier, Vello, and Parley insted of druid-shell and Piet.
+- Integrate AccessKit support.
+- Move the Masonry codebase to the `xilem` repository, ideally in a way that preserves commit history.
+- Port the list of Github Issues in the `masonry-rs` repository to the `xilem` repository.
+- Create a temporary `xilem_masonry` crate, aiming for parity with the current Xilem crate.
+- Once parity is reached, replace the current `src/` folder with the contents of `xilem_masonry`. Remove `xilem_masonry/`.
+- Publish `xilem`.
+- Set the `#masonry` zulip stream's description to a link to the masonry-rs repository.
+
+### How `xilem` source code will be affected
+
+The source code for the `xilem` crate is currently located in the `src/` folder in [the `xilem` repository](https://github.com/linebender/xilem/).
+
+That folder includes:
+
+- **`view/` folder:** Exports reactive-layer Views.
+- **`widget/` folder:** Exports platform-layer Widgets.
+- **`app.rs`:** Exports `App` struct.
+- **`app_main.rs`:** Exports `AppLauncher` struct.
+- **`bloom.rs`:** Defines internal `Bloom` struct.
+- **`geometry.rs`:** Exports `Axis` struct.
+- **`id.rs`:** Exports `Id` type.
+- `lib.rs`
+- **`text.rs`:** Exports `render_text()` function.
+
+All of this source code except for the `view/` folder and `app.rs` will be removed.
+
+Xilem will still export a view trait, defined in `src/view/View.rs` using macros from `xilem_core`. Views will be rewritten to have `masonry::Widget` as their `Element` associated type.
+
+The `app.rs` composition root will be rewritten to be a minimal wrapper around Masonry's composition root. The Masonry composition root essentially takes callbacks when being built; these callbacks will be piped to the Xilem reactive architecture (building a view tree, reconciliation, and events).
```

Comment:
I'm not sure I would go into that detail (callbacks etc.). I think this will clarify in PRs etc. (Because I don't think it will end up just as that, but we'll see).

I think it's enough to just clarify that `app.rs` or `App` (or something along the lines) takes the tasks of building the view tree, reconciliation, and (towards-views) events, and the rest is basically done by masonry which is owned by `app.rs` (however these things communicate, and in whichever threads they may live).